### PR TITLE
flashrom: enable meson build

### PR DESCRIPTION
**Pull request by r-vdp**
This makes it possible to build the flashrom plugin for fwupd.

Ping @jtojnar 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/flashrom/default.nix [hunk 1]
```diff
@@ -20,20 +23,36 @@ stdenv.mkDerivation rec {
...
+  ] ++ lib.optional jlinkSupport libjaylink;
 
-  makeFlags = [ "PREFIX=$(out)" "libinstall" ]
-    ++ lib.optional jlinkSupport "CONFIG_JLINK_SPI=yes";
+  # These options were reworked in the 1.3.0 release,
+  # we'd instead say something like:
+  #   let programmers = lib.concatStringsSep "," ([ "auto" ] ++ optional jlinkSupport [ "_jlink_" ];
+  #   in "-Dprogrammer=${programmers};"
+  mesonFlags = [
+    (lib.strings.mesonBool "-Dconfig_jlink_spi" jlinkSupport)
```
Comment (jtojnar) [inline pkgs/tools/misc/flashrom/default.nix:45]: I did not see `config_jlink_spi` in https://github.com/flashrom/flashrom/blob/v1.2.1/meson_options.txt

**Review by felixsinger — CHANGES_REQUESTED** [review]
The Meson build system in this version is not integrated properly yet and shouldn't be used.

File: pkgs/tools/misc/flashrom/default.nix (hunk 1 above)
Comment (R-VdP) [reply pkgs/tools/misc/flashrom/default.nix:45]: Ah damn, it only got introduced in 1.3.0, not sure how I missed this.

Right, so we'll have to wait for a new release I guess, I'll close this PR, sorry for the noise.

Comment (felixsinger) [reply pkgs/tools/misc/flashrom/default.nix:45]: The Meson build system is not stable with 1.3.0 either.

**Pull request closed without merging**